lib/AST/Decl.cpp

```diff
@@ -4197,12 +4197,20 @@ ParamDecl::ParamDecl(Specifier specifier,
            "'var' cannot appear on parameters; you meant 'inout'");
 }
 
+// Remove the top-level inout type, if there is one.
+static Type removeInOutType(Type type) {
```

Comment:
`getInOutObjectType()` already handles this and I would rather keep usages of `getAs<InOutType>()` centralized.

Reply:
Ugh, how did I miss `getInOutObjectType()`? Search fail, thank you!